src/Aspire.Dashboard/Model/ResourceOutgoingPeerResolver.cs

```diff
@@ -132,11 +132,40 @@ bool TryMatchResourceAddress(string value, [NotNullWhen(true)] out string? name,
         {
             foreach (var (resourceName, resource) in resources)
```

Comment:
TryResolvePeerNameCore has a loop over transformers using the current address. Then there is a lot of functionality inside that loop, which includes looping over transformers again.
 
How about moving logic inside TryMatchResourcesAddress to parse data on the resources to get possible addresses out into an method that returns IEnumerable<string> of possible addresses, and then you loop over the two values with transformers then. That avoids getting and parsing properties from resources as much.

Reply:
Another approach could be caching all the possible endpoints for a resource in this type. That way they aren't recomputed over and over for each span.

Reply:
Call `ExtractResourceAddresses` when a resource changes and store it as a value for _resourceByName together with thre resource view model in a record.